### PR TITLE
Paramedic update

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

Added paramedic's drip that INFRARED_BARON sprited. Could be researched and manufactured in Moe Labs.
Paramedics spawn with all the gear on them, they also get four highcap rubber mags for the pistol.

Helmet will notify wearer if someone's sensors report a health issue. Unless that is turned off via verb.
Scanning interval is 15 seconds, reporting same person again after a minute.

Health monitoring interface also have been reworked, because old one sucked ass and I didn't like it.
Paramedic's tablet is a regular modular tablet with a special "tracking" interaction for monitoring program.

UPD: Armor get high style and slight speed buff, because it's rare, expensive and cool looking but can't have good armor values. A compensation of sort.

<details>
<summary>
Paramedic stuff
</summary>

image

image

image

</details>

<details>
<summary>
Research tree
</summary>

image

image

image

image

image

</details>

<details>
<summary>
Sensor monitoring
</summary>

Tablet:
image

image

image

Console:
image

</details>

<details>
<summary>
Armor and gun stats
</summary>

Vest:
image
Helmet open:
image
Helmet closed:
image

Semiauto:
image
Fullauto:
image

</details>

## Why It's Good For The Game

Cool stuff.

## Changelog
:cl:
add: paramedic armor
add: paramedic's tablet with tracking feature
tweak: Monitoring program now report actual health stats
tweak: research tree expanded with paramedic's update content

/:cl:

### CODE REVIEW

**Comment by TheShown911** [comment]
Armor kind of sucks

**Comment by SirRichardFrancis** [comment]
> Armor kind of sucks

Wdym? Visually or stats?

**Comment by TheShown911** [comment]
Stats wise, but i understand it can't have good stats as a round start armor, maybe give it a 0.3 speed buff so paramedics don't switch off it immediately

File: code/modules/research/nodes/combat.dm [hunk 1]
```diff
@@ -203,16 +204,30 @@
 
 	unlocks_designs = list(/datum/design/research/item/weapon/mindflayer,  /datum/design/research/item/weapon/nuclear, /datum/design/research/item/weapon/lasercannon)
 
+/datum/technology/basic_armor
+	name = "Armor Solutions"
+	desc = "Standart issue armor of moebius paramedic teams."
```
Comment (zombiemimic) [inline code/modules/research/nodes/combat.dm:209]: Should be "Standard"

**Comment by SirRichardFrancis** [comment]
> Stats wise, but i understand it can't have good stats as a round start armor, maybe give it a 0.3 speed buff so paramedics don't switch off it immediately

Done.

File: code/modules/research/nodes/combat.dm (hunk 1 above)
Comment (SirRichardFrancis) [inline code/modules/research/nodes/combat.dm:209]:
```suggestion
	desc = "Standard issue armor of moebius paramedic teams."
```

Comment (SirRichardFrancis) [reply code/modules/research/nodes/combat.dm]: image

**Comment by TheShown911** [comment]
NT has upgrades to their armor while paramedics don't, and they are supposed to stay in their armor for the whole round, not take a random vagabond BP vest for better stats, considering the platinum cost a .3 speed boost shouldn't be too strong

**Comment by Jan-Blasiak** [comment]
Speed buff  would be nice indeed, I would also definitely add a basic health scanner to the helmet.

**Comment by Omni44** [comment]
Upgrades as in better armors they can sometimes print?

**Comment by TheShown911** [comment]
Yes, and their cruciform upgrades.

**Comment by Omni44** [comment]
All of these are rng, and I don't think anyone ever actually managed to use cruciform upgrades sadly. Moebius too can print better armor. I don't wanna get into "Why ur starting armor better than mine" argument, I just don't think we should push the bar for roundstart armors' stats too much. 

Kirov, consider buffing RIG too while at it.

**Comment by TheShown911** [comment]
That's why I told him not to change the stats, but add a slight speed buff so they can do their job better.

File: code/modules/clothing/head/armor.dm [hunk 2]
```diff
@@ -538,3 +538,112 @@
...
+		return
+	
+	if(!speaker_enabled)
+		return
+
+	scan_scheduled = TRUE
+	spawn(scan_interval)
+		if(QDELETED(src))
+			return
+		report_health_alerts()
```
Comment (SirRichardFrancis) [inline code/modules/clothing/head/armor.dm:594]:
```suggestion
		scan_scheduled = FALSE
```

File: code/modules/clothing/head/armor.dm [hunk 3]
```diff
@@ -538,3 +538,112 @@
...
+		scan_scheduled = FALSE
```
Comment (SirRichardFrancis) [inline code/modules/clothing/head/armor.dm:595]:
```suggestion
		report_health_alerts()
```

**Comment by Anonmare** [comment]
Minor nitpick, maybe go for an off-brand name for the pistol to emphasise it's a knock-off?

**Comment by ValoTheValo** [comment]
> Minor nitpick, maybe go for an off-brand name for the pistol to emphasise it's a knock-off?

Its syndicate surplus redone by moebius like the C20-m so it doesnt need to be made by like serbian arms or some no name company. Could just be produced by moebius

File: code/modules/clothing/head/armor.dm [hunk 4]
```diff
@@ -538,3 +538,112 @@
...
+		if(QDELETED(src))
+			return
+		crewmembers_recently_reported.Remove(entry)
+
+
+/obj/item/clothing/head/armor/faceshield/paramedic/proc/report_health_alerts()
+	if(!speaker_enabled)
+		return
+
+	if(!loc || !istype(loc, /mob/living/carbon/human))
```
Comment (MLGTASTICa) [inline code/modules/clothing/head/armor.dm:609]: could just do ishuman(loc)

File: code/modules/clothing/head/armor.dm [hunk 5]
```diff
@@ -538,3 +538,112 @@
...
+		return
+	
+	var/mob/living/carbon/human/user = loc
+
+	var/list/crewmembers = list()
+	var/list/z_levels_to_scan = list(1, 2, 3, 4, 5)
```
Comment (MLGTASTICa) [inline code/modules/clothing/head/armor.dm:615]: you should iterate through all z-levels marked with station Z_level round start , not hard code it.

**Comment by ValoTheValo** [comment]
Why does the paramedic armor cost platinum. Hell it costs even more platinum than the god damn combat voidsuit does

File: code/modules/clothing/head/armor.dm (hunk 4 above)
Comment (SirRichardFrancis) [reply code/modules/clothing/head/armor.dm:609]: Touché.

File: code/modules/clothing/head/armor.dm (hunk 5 above)
Comment (SirRichardFrancis) [reply code/modules/clothing/head/armor.dm:615]: Helmet reacted to mannequin characters, so I set Z-levels to scan this way.

**Comment by SirRichardFrancis** [comment]
> Minor nitpick, maybe go for an off-brand name for the pistol to emphasise it's a knock-off?

Moebius Labs is one of Syndicate's successor corporations, so I guess it's okay for them to manufacture syndicate stuff.

> Why does the paramedic armor cost platinum.

So it can't be printed in large quantities.

> Hell it costs even more platinum than the god damn combat voidsuit does

No it doesn't.

**Comment by Medcar** [comment]
plat cost doesn't make sense for this armor, if you have plat as moebius you're much better off printing the voidsuit instead, making this armor never get used

**Comment by ValoTheValo** [comment]
> > Minor nitpick, maybe go for an off-brand name for the pistol to emphasise it's a knock-off?
> 
> Moebius Labs is one of Syndicate's successor corporations, so I guess it's okay for them to manufacture syndicate stuff.
> 
> > Why does the paramedic armor cost platinum.
> 
> So it can't be printed in large quantities.
> 
> > Hell it costs even more platinum than the god damn combat voidsuit does
> 
> No it doesn't.

Combat Voidsuit costs 3 platinum. this costs 5.

also nobody will print this with the platinum cost. if you have platinum print the combat voidsuit as it is cheaper and more effective.

**Review by TheShown911 — APPROVED** [review]

**Comment by SirRichardFrancis** [comment]
> Combat Voidsuit costs 3 platinum. this costs 5.
> 
> also nobody will print this with the platinum cost. if you have platinum print the combat voidsuit as it is cheaper and more effective.

Combat voidsuit cost 5, this cost 5 as well. Don't forget that upgraded lathe reduce printing cost.

**Comment by SirRichardFrancis** [comment]
image

**Comment by Kegdo** [comment]
@SirRichardFrancis unconflict when?

File: code/modules/clothing/head/armor.dm [hunk 6]
```diff
@@ -543,3 +543,113 @@
...
+	set src in usr
+
+	if(speaker_enabled)
+		to_chat(usr, SPAN_WARNING("[src] beeps: 'Notifications disabled.'"))
+		speaker_enabled = FALSE
+	else
+		to_chat(usr, SPAN_WARNING("[src] beeps: 'Notifications enabled.'"))
+		speaker_enabled = TRUE
+		report_health_alerts()
+		schedule_scan()
```
Comment (SirRichardFrancis) [inline code/modules/clothing/head/armor.dm:655]:
```suggestion

```

**Comment by TheShown911** [comment]
Merg this PR.

**Comment by hyperioo** [comment]
Sorry ;-;

**Comment by Humonitarian** [comment]
Add tint_down = TINT_LOW to the helmet if you want less obscuration. Do note the obscuration will be small enough to be easy to ignore and make face_up mostly pointless beyond RP.

**Comment by TheShown911** [comment]
BRUH

**Comment by drexample** [comment]
BRUH

**Comment by SirRichardFrancis** [comment]
BRUH indeed.

**Comment by SirRichardFrancis** [comment]
Conflict-free again.

**Comment by ACCount12** [comment]
In the future, do try to split your PRs. Armor in one PR, sensor monitor in another, etc.

Bundling lots of tangentially related shit in a single "update" PR makes reviewing them a mess for no good reason.

**Comment by SirRichardFrancis** [comment]
> In the future, do try to split your PRs. Armor in one PR, sensor monitor in another, etc.
> Bundling lots of tangentially related shit in a single "update" PR makes reviewing them a mess for no good reason.

This stuff is related by a common theme and modifying same files.
By splitting it into several PRs I either will get guaranteed conflicts or will have to make more PRs after previous is merged, delaying one update for weeks or months.

**Pull request merged**